lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+        }
+
+        /**
+         * Checks that the user-intended non-base ten number equals the actual number after is has been converted to the Number type
+         * @param {Node} node the node being evaluated
+         * @returns {boolean} true if they do not match
+         */
+        function checkNotBaseTenNumber(node) {
+            const rawString = node.raw.toUpperCase();
+            let base = 0;
+
+            if (rawString.startsWith("0B")) {
+                base = 2;
+            } else if (rawString.startsWith("0X")) {
+                base = 16;
+            } else {
+                base = 8;
+            }
+
+            return !rawString.endsWith(node.value.toString(base).toUpperCase());
+        }
+
+        /**
+         * Adds a decimal point to the numeric string at index 1
+         * @param {string} stringNumber the numeric string without any decimal point
+         * @returns {string} the numeric string with a decimal point in the proper place
+         */
+        function addDecimalPointToNumber(stringNumber) {
+            return `${stringNumber.slice(0, 1)}.${stringNumber.slice(1)}`;
+        }
+
+        /**
+         * Returns the number stripped of leading zeros
+         * @param {string} numberAsString the string representation of the number
+         * @returns {string} the stripped string
+         */
+        function removeLeadingZeros(numberAsString) {
+            return numberAsString.replace(/^0*/u, "");
+        }
+
+        /**
+         * Returns the number stripped of trailing zeros
+         * @param {string} numberAsString the string representation of the number
+         * @returns {string} the stripped string
+         */
+        function removeTrailingZeros(numberAsString) {
+            return numberAsString.replace(/0*$/u, "");
+        }
+
+        /**
+         * Converts an integer to to an object containing the the integer's coefficient and order of magnitude
+         * @param {string} stringInteger the string representation of the integer being converted
+         * @returns {Object} the object containing the the integer's coefficient and order of magnitude
+         */
+        function normalizeInteger(stringInteger) {
+            const significantDigits = removeTrailingZeros(stringInteger);
+
+            return {
+                magnitude: stringInteger.length - 1,
+                coefficient: addDecimalPointToNumber(significantDigits)
+            };
+        }
+
+        /**
+         *
+         * Converts a float to to an object containing the the floats's coefficient and order of magnitude
+         * @param {string} stringFloat the string representation of the floate being converted
+         * @returns {Object} the object containing the the integer's coefficient and order of magnitude
+         */
+        function normalizeFloat(stringFloat) {
+            if (stringFloat.startsWith("0") || stringFloat.startsWith(".")) {
```

Comment:
This will handle literals like `019.5` incorrectly, because only the `5` will be treated as a significant digit.